src/extension.ts

```diff
@@ -621,6 +625,24 @@ function installCustomRequestHandlers(context: VSCode.ExtensionContext) {
     await hotspotsTreeDataProvider.refresh(hotspotsPerFile);
     updateSonarLintViewContainerBadge();
   });
+  languageClient.onNotification(protocol.PublishTaintVulnerabilitiesForFile.type, async taintVulnerabilitiesPerFile => {
+    const diagnostics = taintVulnerabilitiesPerFile.diagnostics.map(diagnostic => {
+      const d = new VSCode.Diagnostic(
+        new VSCode.Range(
+          new VSCode.Position(diagnostic.range.start.line, diagnostic.range.start.character),
+          new VSCode.Position(diagnostic.range.end.line, diagnostic.range.end.character)
+        ),
+        diagnostic.message,
+        DiagnosticSeverity.Error
```

Comment:
AFAICT we only ever publish `Warning` or `Hint` level diagnostics; and we should probably use the one sent by the LS, for consistency and support for "on new code".
```suggestion
        diagnostic.severity
```